### PR TITLE
v1.0 proposal draft

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Draft

[Formatted version](https://github.com/weaveworks/eksctl/blob/v1-proposal/docs/proposal-007-v1-changes.md)
### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [ ] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

**Comment by michaelbeaumont** [comment]
Should we tag the issues we think should be part of 1.0?

**Comment by aclevername** [comment]
> Should we tag the issues we think should be part of 1.0?

sounds good! I've create the `area/v1.0` label

File: docs/proposal-007-v1-changes.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## Goals
+- TODO
+
+## Non-Goals
+- TODO
+## Risks
+- Adoption of the new CLI version will be slow
+
+- Users will dislike migrating config file format
```
Comment (Callisto13) [inline docs/proposal-007-v1-changes.md:47]: Probably a mitigation would be to do something fancy for them.  like we provide a translate subcommand

File: docs/proposal-007-v1-changes.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+- Community will ask us to maintain the older version
```
Comment (Callisto13) [inline docs/proposal-007-v1-changes.md:49]: we'll need a nice clear timeline and be very strict on when we move from "actively developed" -> "maintenance" -> "EOL"

File: docs/proposal-007-v1-changes.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+This proposal outlines a list of features and breaking changes to be included in the v1.0 release of eksctl. This
+proposal can be used as a place to reference other proposal for changes and should serve as an overarching document to
+see what will change in v1.0.
+
+### Draft list of changes
+This is a draft list of changes that could be made in a v1.0 release. Some of these may required their own proposals.
+
+- Introduction of [eksctl apply](https://github.com/weaveworks/eksctl/pull/3037)
+
+- Removal of deprecated functionality
```
Comment (Callisto13) [inline docs/proposal-007-v1-changes.md:15]: with `apply` will all current subcommands and flags disappear?

**Comment by michaelbeaumont** [comment]
Can't we just discuss everything on the issues themselves and tag them with `v1.0`? I.e. everything in this proposal is necessarily scoped to one issue.

**Comment by Callisto13** [comment]
ah yeh you are right. we should use this proposal as just a grouping doc

**Comment by aclevername** [comment]
> ah yeh you are right. we should use this proposal as just a grouping doc

I do propose that in the intro 😉  https://github.com/weaveworks/eksctl/pull/3047/files#diff-dc3c816fca42f7d3a40a3181f94cd419e321499ba38b0922b2b4be954046209dR6-R8

**Comment by michaelbeaumont** [comment]
Do we need the proposal at all? What is there to propose about `v1.0` itself? If we assume it exists then it's defined by the issues that make it into `v1.0`.

**Comment by Callisto13** [comment]
> I do propose that in the intro

Expecting me to actually read stuff was your first mistake

**Comment by Callisto13** [comment]
> Do we need the proposal at all? What is there to propose about v1.0 itself? If we assume it exists then it's defined by the issues that make it into v1.0.

I like having a "thing" to group and link to for when people ask about v1, don't really mind what form it takes.

**Comment by aclevername** [comment]
> > Do we need the proposal at all? What is there to propose about v1.0 itself? If we assume it exists then it's defined by the issues that make it into v1.0.
> 
> I like having a "thing" to group and link to for when people ask about v1, don't really mind what form it takes.

+ 1

I created it initially because we had a small mental list of things/break changing to go into `v1.0` and they were not being tracked. This proposal or another solution for tracking them and providing a space for community discussion is all we need really

**Pull request closed without merging**